Integrations/PhishLabsIOC_EIR/PhishLabsIOC_EIR.py

```diff
@@ -271,55 +288,76 @@ def test_module_command(client: Client, *_) -> Tuple[None, None, str]:
 def fetch_incidents_command(
         client: Client,
         fetch_time: str,
-        last_run: Optional[str] = None) -> Tuple[List, str]:
+        limit: str,
+        last_run: Optional[str] = None) -> Tuple[List[Dict[str, Any]], Optional[str]]:
     """Uses to fetch incidents into Demisto
     Documentation: https://github.com/demisto/content/tree/master/docs/fetching_incidents
 
     Args:
         client: Client object with request
         fetch_time: From when to fetch if first time, e.g. `3 days`
+        limit: limit of incidents in a fetch
         last_run: Last fetch object occurs.
 
     Returns:
         incidents, new last_run
     """
+    # Init
+    raws: List = []
+    incidents_raw: List = []
+    # Set last run time
     occurred_format = '%Y-%m-%dT%H:%M:%SZ'
-    # Get incidents from API
-    if not last_run:  # if first time running
-        datetime_new_last_run, _ = parse_date_range(fetch_time, date_format=occurred_format)
+    if not last_run:
+        datetime_new_last_run, _ = parse_date_range(date_range=fetch_time,
+                                                    date_format=occurred_format)
     else:
-        datetime_new_last_run = parse_date_string(last_run)
-    new_last_run = datetime_new_last_run.strftime(occurred_format)
-    kwargs = {
-        'created_after': new_last_run,
-    }
-    raws: List = []
-    incidents: List = []
-    offset = 0
-    raw_response = client.phishlabs_ioc_eir_get_incidents(**kwargs)
-    while raw_response.get('metadata', {}).get('count'):
+        datetime_new_last_run = last_run
+    # Query incidents by limit and creation time
+    total = 0
+    offset = 50
+    limit_incidents = int(limit)
+    limit_page = 50
+    if limit_incidents <= 50:
+        limit_page = limit_incidents
+    raw_response = client.get_incidents(created_after=datetime_new_last_run,
+                                        offset=offset,
+                                        limit=limit_page,
+                                        sort='created_at',
+                                        direction='asc')
+    while raw_response.get('metadata', {}).get('count') and total < limit_incidents:
         raws.append(raw_response)
-        incidents += raw_response.get('incidents', [])
-        offset += 50
-        raw_response = client.phishlabs_ioc_eir_get_incidents(offset=offset, **kwargs)
-    if incidents:
-        for incident in incidents:
+        incidents_raw += raw_response.get('incidents', [])
+        total += int(raw_response.get('metadata', {}).get('count'))
+        if total >= limit_incidents:
+            break
+        if limit_incidents - total < 50:
+            limit_page = limit_incidents - total
+        offset += limit_page
```

Comment:
shouldn't offset be updated with the length of the last fetched data (`raw_response.get('metadata', {}).get('count')`)?

Reply:
You right it, it can be a problem if I don't get the expected amount.